src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+    This model is a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass. Use it
+    as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage and
+    behavior.
+
+    Parameters:
+        config ([`MaskRCNNConfig`]): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+MASK_RCNN_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See
+            [`AutoImageProcessor.__call__`] for details.
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    """
+    Mask R-CNN Model (consisting of a backbone, region-proposal network (RPN) and RoI head) for object detection and
+    instance segmentation tasks.
+    """,
+    MASK_RCNN_START_DOCSTRING,
+)
+class MaskRCNNForObjectDetection(MaskRCNNPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.config = config
+
+        self.backbone = AutoBackbone.from_config(config.backbone_config)
+        self.neck = MaskRCNNFPN(config, self.backbone.channels)
+        self.rpn_head = MaskRCNNRPN(config)
+        self.roi_head = MaskRCNNRoIHead(config)
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def parse_losses(self, losses):
+        """Parse the raw outputs (losses) of the network.
+
+        Source:
+        https://github.com/open-mmlab/mmdetection/blob/ff9bc39913cb3ff5dde79d3933add7dc2561bab7/mmdet/models/detectors/base.py#L176
+
+        Args:
+            losses (`dict`):
+                Raw output of the network, which usually contain losses and other necessary information.
+
+        Returns:
+            `torch.Tensor`: The overall loss which is a weighted sum of all losses.
+        """
+        log_vars = OrderedDict()
+        for loss_name, loss_value in losses.items():
+            if isinstance(loss_value, torch.Tensor):
+                log_vars[loss_name] = loss_value.mean()
+            elif isinstance(loss_value, list):
+                log_vars[loss_name] = sum(_loss.mean() for _loss in loss_value)
+            else:
+                raise TypeError(f"{loss_name} is not a tensor or list of tensors")
+
+        loss = sum(_value for _key, _value in log_vars.items() if "loss" in _key)
+
+        return loss
```

Comment:
We can skip some the iteration logic

```suggestion
        loss = 0
        for loss_name, loss_value in losses.items():
            if "loss" not in loss_name: 
                continue
            if isinstance(loss_value, torch.Tensor):
                loss += loss_value.mean()
            elif isinstance(loss_value, list):
                for _loss in loss_value:
                    loss += _loss.mean()
            else:
                raise TypeError(f"{loss_name} is not a tensor or list of tensors")

        return loss
```